src/proj.h

```diff
@@ -672,7 +672,7 @@ PJ_COORD PROJ_DLL proj_geod (const PJ *P, PJ_COORD a, PJ_COORD b);
 #define PROJ_ERR_OTHER_NETWORK_ERROR                     (PROJ_ERR_OTHER+3)             /* failure when accessing a network resource */
 
 /* Set or read error level */
-int  PROJ_DLL proj_context_errno (PJ_CONTEXT *ctx);
+int  PROJ_DLL proj_context_errno (const PJ_CONTEXT *ctx);
```

Comment:
As far as I can tell this is the only slightly concerning change. I'm unsure if we can keep the `const` promise but we probably can. I think we've talked about doing something similar before (for other objects?) without really getting to the bottom of it. 

What are the consequences of this going forwards? If any. I suspect downstream code will run just fine in most cases but I'm unsure if it holds true in *all* cases.

Reply:
Doesn't it impact ABI which implies a soversion change?
If it does, must this change target a particular PROJ version change?

Reply:
> Doesn't it impact ABI which implies a soversion change?

That's what I'm thinking but I don't fully understand the implications. If this breaks things we'll have to go to version 10 and that's at least five years into the future.

Reply:
This change is safe from an ABI point of view. The constness of arguments is not captured in the C ABI (contrary to C++ symbols). And from an API point of view, changing from ``foo*`` to ``const foo*`` is also safe as it doesn't trigger any compiler warnings.

Reply:
Thanks for clearing that up